plugins/lightspeed/src/api/LightspeedApiClient.ts

```diff
@@ -0,0 +1,154 @@
+import { ConfigApi, IdentityApi } from '@backstage/core-plugin-api';
+
+import { LightspeedAPI } from './api';
+
+export type Options = {
+  configApi: ConfigApi;
+  identityApi: IdentityApi;
+};
+
+export class LightspeedApiClient implements LightspeedAPI {
+  private readonly configApi: ConfigApi;
+  private readonly identityApi: IdentityApi;
+
+  constructor(options: Options) {
+    this.configApi = options.configApi;
+    this.identityApi = options.identityApi;
+  }
+
+  async getBaseUrl() {
+    return `${this.configApi.getString('backend.baseUrl')}/api/lightspeed`;
+  }
+
+  getServerUrl() {
+    // Currently supports a single llm server
+    return `${this.configApi.getConfigArray('lightspeed.servers')[0].getOptionalString('url')}`;
+  }
+
+  async getUserAuthorization() {
+    const { token: idToken } = await this.identityApi.getCredentials();
+    return idToken;
+  }
+
+  async createMessage(
+    prompt: string,
+    selectedModel: string,
+    conversation_id: string,
+  ) {
+    const baseUrl = await this.getBaseUrl();
+    const token = await this.getUserAuthorization();
+
+    const response = await fetch(`${baseUrl}/v1/query`, {
```

Comment:
can just use the fetchAPI backstage provided.  for example:
```
  import { FetchApi
} from '@backstage/core-plugin-api';

... ...
export type Options = {
  configApi: ConfigApi;
  identityApi: IdentityApi;
  fetchApi: FetchApi;
};

... ...

        const resp = await this.fetchApi.fetch(`${baseUrl}/conversations`, {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json',
          },
        });

```

according to https://backstage.io/docs/auth/#identity-for-plugin-developers, frontend plugin should call backend plugin API via the fetchAPI, and by-default, the login information will be appended as part of the call